rts/Rendering/Env/Decals/LegacyTrackHandler.cpp

```diff
@@ -421,6 +421,8 @@ void LegacyTrackHandler::AddTrack(const CUnit* unit, const float3& newPos)
 		return;
 	if (!unitDef->IsGroundUnit())
 		return;
+	if (unitDef->floatOnWater && newPos[1] <= unitDef->waterline * -1.0f)
```

Comment:
This can be reduced to `if (unit->IsInWater() && !unit->IsOnGround())`, a unit's waterline value might change after it is loaded.

Reply:
Thanks @rtri, that's much better.